packages/utils/src/dal/mikro-orm/integration-tests/__tests__/mikro-orm-repository.spec.ts

```diff
@@ -0,0 +1,844 @@
+    }
+  }
+}
+
+const Entity1Repository = mikroOrmBaseRepositoryFactory<Entity1>(Entity1)
+const Entity2Repository = mikroOrmBaseRepositoryFactory<Entity2>(Entity2)
+const Entity3Repository = mikroOrmBaseRepositoryFactory<Entity3>(Entity3)
+
+describe("mikroOrmRepository", () => {
+  describe("upsert with replace", () => {
+    let orm!: MikroORM
+    let manager!: EntityManager
+    const manager1 = () => {
+      return new Entity1Repository({ manager: manager.fork() })
+    }
+    const manager2 = () => {
+      return new Entity2Repository({ manager: manager.fork() })
+    }
+    const manager3 = () => {
+      return new Entity3Repository({ manager: manager.fork() })
+    }
+
+    beforeEach(async () => {
+      await dropDatabase(
+        { databaseName: DB_NAME, errorIfNonExist: false },
+        pgGodCredentials
+      )
+
+      orm = await MikroORM.init({
+        entities: [Entity1, Entity2],
+        clientUrl: getDatabaseURL(),
+        type: "postgresql",
+      })
+
+      const generator = orm.getSchemaGenerator()
+      await generator.ensureDatabase()
+      await generator.createSchema()
+
+      manager = orm.em.fork()
+    })
+
+    afterEach(async () => {
+      const generator = orm.getSchemaGenerator()
+      await generator.dropSchema()
+      await orm.close(true)
+    })
+
+    it("should successfully create a flat entity", async () => {
+      const entity1 = { id: "1", title: "en1" }
+
+      const resp = await manager1().upsertWithReplace([entity1])
+      const listedEntities = await manager1().find()
+
+      expect(listedEntities).toHaveLength(1)
+      expect(listedEntities[0]).toEqual(
+        expect.objectContaining({
+          id: "1",
+          title: "en1",
+        })
+      )
+    })
+
+    it("should successfully update a flat entity", async () => {
+      const entity1 = { id: "1", title: "en1" }
+
+      await manager1().upsertWithReplace([entity1])
+      entity1.title = "newen1"
+      await manager1().upsertWithReplace([entity1])
+      const listedEntities = await manager1().find()
+
+      expect(listedEntities).toHaveLength(1)
+      expect(listedEntities[0]).toEqual(
+        expect.objectContaining({
+          id: "1",
+          title: "newen1",
+        })
+      )
+    })
+
+    it("should successfully do a partial update a flat entity", async () => {
+      const entity1 = { id: "1", title: "en1" }
+
+      await manager1().upsertWithReplace([entity1])
+      entity1.title = undefined as any
+      await manager1().upsertWithReplace([entity1])
+      const listedEntities = await manager1().find()
+
+      expect(listedEntities).toHaveLength(1)
+      expect(wrap(listedEntities[0]).toPOJO()).toEqual(
+        expect.objectContaining({
+          id: "1",
+          title: "en1",
+        })
+      )
+    })
+
+    // TODO: Should we support this
+    it("should successfully create an entity with a sub-entity many-to-one relation", async () => {
```

Comment:
In my opinion, we shouldn't allow this. But this is because I can't see a use case for it - I fint it off that a child can create its parent. If anyone has a good one, I am OK with keeping it in.

Reply:
no this is true, a child should not create its parent, here it would mean a fulfillment can create a shipping option, or a variant can create a product. I don't believe it should be feasable

Reply:
only the following type of relation should create
- many to many
- one to one
- one to many

Reply:
@olivermrbl I was on the fence about this one, and I initially implemented without supporting it. We have a couple of places where this is currently possible in the product module, and since what to upsert needs to be explicitly stated, I thought it won't harm us if we support it. I can remove it if you think that's better, of course.

Reply:
I believe we should not support that, furthermore, with the new conventions around relations, all many to one are marked as persist false as we are only supposed to attach the foreign key but the parent is expected to already exists and the child being either created separately or through the one to many cascade on the collection

Reply:
Yeah, I would also remove the support for it

Reply:
Ok sounds good 👍